prov/rxm/src/rxm_conn.c

```diff
@@ -136,59 +153,82 @@ static void rxm_send_queue_close(struct rxm_send_queue *send_queue)
 static void rxm_conn_close(struct util_cmap_handle *handle)
 {
 	struct rxm_conn *rxm_conn = container_of(handle, struct rxm_conn, handle);
+	struct rxm_ep *rxm_ep = container_of(handle->cmap->ep,
+					     struct rxm_ep, util_ep);
 
 	if (!rxm_conn->msg_ep)
 		return;
-	/* Save the preposted RX list for further cleanup */
-	dlist_splice_tail(&rxm_conn->saved_posted_rx_list,
+
+	rxm_conn->close_data = calloc(1, sizeof(*rxm_conn->close_data));
+	if (!rxm_conn->close_data) {
+		assert(0);
+		rxm_conn_close_msg_ep(rxm_ep, rxm_conn);
+		return;
+	}
+
+	rxm_conn->close_data->msg_ep = rxm_conn->msg_ep;
```

Comment:
ofi_cmap_process_connreq() in util_av.c seems to be calling this rxm_conn_close()  with the rxm_conn->msg_ep is for which the connection is accepted. So when the sent out connreq is refused by peer we are closing this msg_ep with valid connection. Am I missing something here?